wallet-new/src/Cardano/Wallet/Kernel/DB/Spec.hs

```diff
@@ -268,23 +324,73 @@ instance IsCheckpoint PartialCheckpoint where
     cpBlockMeta   = pcheckpointBlockMeta
     cpForeign     = pcheckpointForeign
     cpContext     = pcheckpointContext
+    delta         = deltaPartialC
+    step          = stepPartialC
 
 cpAddressMeta :: IsCheckpoint c => Core.Address -> Lens' c AddressMeta
 cpAddressMeta addr = cpBlockMeta . _Wrapped . addressMeta addr
 
+{-------------------------------------------------------------------------------
+  Convenience: definitions for compression
+-------------------------------------------------------------------------------}
+
+deltaC :: Checkpoint -> Checkpoint -> DeltaCheckpoint
```

Comment:
`deltaCheckpoint` is longer but I think it clarifies a lot 😁 

Alternatively, what about just dropping the suffix and use `delta` alone. Yes, it means you will have to import this qualified somewhere as the name will conflict with something else, but I find more readable something like:

```
c1 `Checkpoint.delta` c2
```

Than

```
c1 `deltaC` c2
```

But mind, this is my personal preference. Folks like @edsko or other chaps in the company might think the exact opposite 😂 

🚴